src/Storages/StorageXDBC.cpp

```diff
@@ -184,22 +185,42 @@ namespace
         {
             ASTs & engine_args = args.engine_args;
 
-            if (engine_args.size() != 3)
-                throw Exception(ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH,
-                    "Storage {} requires exactly 3 parameters: {}('DSN', database or schema, table)", name, name);
-
-            for (size_t i = 0; i < 3; ++i)
-                engine_args[i] = evaluateConstantExpressionOrIdentifierAsLiteral(engine_args[i], args.getLocalContext());
+            String connection_string;
+            String database_or_schema;
+            String table;
+
+            if (auto named_collection = tryGetNamedCollectionWithOverrides(engine_args, args.getLocalContext()))
+            {
+                connection_string = name == "JDBC" ? named_collection->get<String>("datasource") : named_collection->get<String>("connection_settings");
+                table = name == "JDBC" ? named_collection->get<String>("table") : named_collection->get<String>("external_table");
+                database_or_schema = name == "JDBC" ? named_collection->get<String>("schema") : named_collection->get<String>("external_database");
```

Comment:
Not very nice, but seems acceptable to me.
The only what I suggest is keeping the same order (table consistently before or consistently after database_or_schema).